src/core/processing/qgsprocessingparameterdxflayers.cpp

```diff
@@ -61,16 +63,22 @@ bool QgsProcessingParameterDxfLayers::checkValueIsAcceptable( const QVariant &in
     const QVariantList layerList = input.toList();
     for ( const QVariant &variantLayer : layerList )
     {
-      if ( qobject_cast< QgsVectorLayer * >( qvariant_cast<QObject *>( variantLayer ) ) )
-        continue;
+      vectorLayer = qobject_cast< QgsVectorLayer * >( qvariant_cast<QObject *>( variantLayer ) );
+      if ( vectorLayer )
+      {
+        if ( vectorLayer->isSpatial() )
+          continue;
+        else
+          return false;
```

Comment:
if only 1 layer isn't spatial, wouldn't it be better to just skip it rather than discarding the whole list?

Reply:
For me that would make sense as well, but here I'm following the semantics of the `checkValueIsAcceptable()` method when the input is a list. As I see in other implemented processing parameters, it's either all elements are valid or none. 

See for instance: [qgsprocessingparameteralignrasterlayers](https://github.com/qgis/QGIS/blob/9e0a6761e518387316317ecda299284af5f0827e/src/core/processing/qgsprocessingparameteralignrasterlayers.cpp#L74-L75), [qgsprocessingparameterfieldmap](https://github.com/qgis/QGIS/blob/9e0a6761e518387316317ecda299284af5f0827e/src/core/processing/qgsprocessingparameterfieldmap.cpp#L50-L55), or  [qgsprocessingparametervectortilewriterlayers](https://github.com/qgis/QGIS/blob/9e0a6761e518387316317ecda299284af5f0827e/src/core/processing/qgsprocessingparametervectortilewriterlayers.cpp#L63-L64).